support/tee_call/src/lib.rs

```diff
@@ -69,6 +69,8 @@ pub trait TeeCall: Send + Sync {
     fn supports_get_derived_key(&self) -> Option<&dyn TeeCallGetDerivedKey>;
     /// Get the [`TeeType`].
     fn tee_type(&self) -> TeeType;
+    /// Support clone for [`Box<dyn TeeCall>`]
+    fn clone_box(&self) -> Box<dyn TeeCall>;
```

Comment:
What's the actual reason we have this clone_box requirement floating around?

i.e: what code depends on the trait object being clonable?

it might be possible to strip out all of this infrastructure, simply by swapping out some of the currnet uses of `Box<dyn TeeCall>` for `Arc<dyn TeeCall>`, given that all the current TeeCall methods take `&self` instead of `&mut self`

Reply:
We need clone the `TpmRequestAkCertHelper` for support async request:
https://github.com/microsoft/openvmm/blob/5492803f6c0f9f051e4413ecce18ac6835238460/vm/devices/tpm/src/lib.rs#L814

Reply:
I see.

Yeah, if that's the only use-case here, I would suggest swapping all those Box-es over for Arc-s, and then this can just be a regular .clone